### DISCUSSION TITLE
Tablist Overhaul

### DISCUSSION THREAD

**Discussion by SenseiMunchkin** [Zrips/CMI#5769] [Feature Requests]
I use CMI as much as I can, but one feature I find to be lacking in features is the Tablist feature. I've always had to disable that module in favor of another plugin, for the main reason being that other plugins allow players to be sorted. If this could be added to CMI, I'd be very grateful and I'm sure other purchasers of the plugin would be as well. Three types of sorting:

# Group Sorting
(typically free in other plugins)

Sorting by group is the most popular among other tablist plugins, so it makes sense to add it here. All it does it sort by group in an ascending order using a list. TAB [here](https://www.spigotmc.org/resources/tab-1-5-x-1-17-free-version.57806/): 
![image](https://user-images.githubusercontent.com/25995812/123669162-6bcb8900-d801-11eb-9a10-b4732d0228f4.png)
Something like this would be all that's necessary for this type of sorting.

# Permission Sorting
(typically free in other plugins)

Sorting by permission is a bit difficult for server owners, since most don't set their permissions properly and OP all their staff, but for the experienced owner this could be a useful feature. Again, ascending order. (Example same as Group Sorting, just substitute groups with permissions).

# Placeholder Sorting
(paid feature in other plugins)

Sorting by placeholder is my absolute favorite, since most of my work involves custom placeholders that make the visuals great looking. This would come in handy for sorting by player level, player faction/towny names, and many more values. In order for this to work, a sorting rule would be set. Either: `a-z` (alphabetical order), `z-a`(reverse alphabetical order), '0-9'(ascending numerical order), '9-0' (descending numerical order). 

# Aligning Suffixes on the Right
(paid feature in other plugins)

An example is shown below:
![image](https://user-images.githubusercontent.com/25995812/123671375-d087e300-d803-11eb-9bc4-633651bcaedb.png)
A cool feature to have to potentially make the tablist more uniform, depending on that the user wants to display.

EDIT: added clarity to the text under the headers
EDIT 2: added some more requested features, and renamed title to match accordingly

**Comment by mrfloris** [comment]
Tablist.yml has player-group sorting in the latest builds of cmilib and cmi. Upgrade and review tablist.yml's configuration (bottom of page). You can sort ascending or descending by type, and types range from permission node weight to baltop rank, etc.
